bindings/python/bt2/bt2/fields.py

```diff
@@ -111,6 +111,9 @@ def __float__(self):
         return float(self._value)
 
     def __repr__(self):
+        if not self.is_set:
```

Comment:
Since _is set_ is a property of `_Field`, why not use a template method here:

    class _Field(object._Object, metaclass=abc.ABCMeta):
        # ...

        def _repr(self):
            raise NotImplementedError

        def __repr__(self):
            return self._repr() if self.is_set else 'Unset'

        # ...

and replace each specific field class's `__repr__()` with `_repr()`, and remove the redundant `if not self.is_set` check.